File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Applications/WebSockets/WebSocketsSampleApps/CSharp/CompanionKitWebSocketMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

using Gizmox.WebGUI;
using Gizmox.WebGUI.Forms;
using Newtonsoft.Json;

namespace Gizmox.WebGUI.Forms.CompanionKit.WebSocketsSampleAppsCS
{
    public class CompanionKitWebSocketMessage
    {
        /// <summary>
        /// Gets or sets the type of the object.
        /// </summary>
        /// <value>
        /// The type of the object.
        /// </value>
        public Type ObjectType { get; set; }

        public CompanionKitWebSocketMessage()
        {
            this.ObjectType = this.GetType();
        }

        /// <summary>
        /// Sends the message.
        /// </summary>
        /// <param name="objMessage">The obj message.</param>
        public static void SendMessage(CompanionKitWebSocketMessage objMessage)
        {
            //Serialize object to string 
            string strMessage = JsonConvert.SerializeObject(objMessage);
            //Sends serialized message via WebSockets
            Application.WebSockets.Send(Application.WebSockets.ConnectionId, strMessage, SendType.All);
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Concepts/WebSockets/DynamicData/CompanionKitWebSocketMessage.vb
Imports Newtonsoft.Json
Imports Gizmox.WebGUI.Forms

Namespace WebSocketsSampleAppsVB
    Public Class CompanionKitWebSocketMessage
        ''' <summary>
        ''' Gets or sets the type of the object.
        ''' </summary>
        ''' <value>
        ''' The type of the object.
        ''' </value>
        Public Property ObjectType() As Type
            Get
                Return m_ObjectType
            End Get
            Set(ByVal value As Type)
                m_ObjectType = value
            End Set
        End Property
        Private m_ObjectType As Type

        Public Sub New()
            Me.ObjectType = Me.[GetType]()
        End Sub

        ''' <summary>
        ''' Sends the message.
        ''' </summary>
        ''' <param name="objMessage">The obj message.</param>
        Public Shared Sub SendMessage(ByVal objMessage As CompanionKitWebSocketMessage)
            'Serialize object to string 
            Dim strMessage As String = JsonConvert.SerializeObject(objMessage)
            'Sends serialized message via WebSockets
            Application.WebSockets.Send(Application.WebSockets.ConnectionId, strMessage, SendType.All)
        End Sub
    End Class
End Namespace